lib/src/facts/posix/uptime_resolver.cc

```diff
@@ -43,6 +48,22 @@ namespace facter { namespace facts { namespace posix {
 
     int64_t uptime_resolver::get_uptime()
     {
+        // We first try to get the system boot time by reading it directly from
+        // the utmp file using the getutxid system call.
+        LOG_DEBUG("Attempting to calculate the uptime from the utmp file...");
+        struct utmpx query;
+        query.ut_type = BOOT_TIME;
+        setutxent();
+        struct utmpx* ent = getutxid(&query);
```

Comment:
It's not really idiomatic C++ to use `struct` when declaring variables. You'll see it in manpages/docs because it's necessary syntax in C, but C++ treats structs as first-class types.